pir/proto/payload.proto

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright 2020 the authors listed in CONTRIBUTORS.md
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+syntax = "proto2";
+
+package pir;
+/*
+#include "pir/proto/pryload.pb.h"
+*/
+
+/*PayloadData holds the request/response data buffers, usually ciphertexts */
+message PayloadData {
+  repeated bytes data = 1; //query or response data
+}
+
+/*Payload contains the entire request or response message */
+message Payload {
+  required PayloadData data = 1; //payload data
```

Comment:
Trust me, you do _not_ want to set required here. From https://developers.google.com/protocol-buffers/docs/proto 

> Required Is Forever You should be very careful about marking fields as required. If at some point you wish to stop writing or sending a required field, it will be problematic to change the field to an optional field – old readers will consider messages without this field to be incomplete and may reject or drop them unintentionally. You should consider writing application-specific custom validation routines for your buffers instead. Some engineers at Google have come to the conclusion that using required does more harm than good; they prefer to use only optional and repeated. However, this view is not universal.

From my point of view, required is basically _never_ needed and usually does more harm than good. It's more important to validate messages you receive than try to have the library do that for you, since any change will either break backwards compatibility or break your validation. This is a lesson learned after building _many_ systems at Google.